### ISSUE TITLE
what is cu_input_lens in the modeling_llama.py ?

### ISSUE THREAD

**Issue by lixali** [bytedance/HLLM#19]
In https://github.com/bytedance/HLLM/blob/2deed176960901aa88a8864e6f6902a194c4e970/code/REC/model/HLLM/modeling_llama.py#L1035 , what is "cu_input_lens" and is it related to the attention_mask? 

And why "cu_input_lens" can be None?

**Comment by ssyzeChen** [comment]
`cu_input_lens` is used in flash attention for faster training and inference by remove padding. When `cu_input_lens` is None, there are some other ways to implement rmpad by https://github.com/bytedance/HLLM/blob/main/code/REC/model/HLLM/flash_self_attn.py#L92. Due to the large number of items (i.e., large batch size) in the Item LLM, this method is less efficient. Therefore, we choose to directly obtain cu_input_lens during data preprocessing.
